### ISSUE TITLE
BUG to run bpa model

### ISSUE THREAD

**Issue by matiirueda** [microsoft/semantic-link-labs#78]
**Describe the bug**
I’m trying to run the BPA with the following script:
WorskpaceSemanticModelName = 'Riesgo Operacional y Reputacional'
df_datasets = fabric.list_datasets(WorskpaceSemanticModelName)
for index, row in df_datasets.iterrows():
    mydataset = row['Dataset Name']
    if (mydataset != 'VF_Auditoria_Modelos') and (mydataset != 'BPA_Test'):
        try:
            labs.run_model_bpa(dataset=mydataset, workspace=WorskpaceSemanticModelName, export=True)
        except Exception as e:
            print(f"Error processing {mydataset}: {e}")
            continue
However, it returns the following error for all cases:

Error processing Perdidas Mensual Delta: There's no item with the ID '4916c29a-88ba-40f4-8e16-e28d878593b6' in workspace 'Riesgo Operacional y Reputacional'
...
It seems that the script is incorrectly assigning the same ID to all datasets, which is causing the correct items not to be found. The df_datasets seems to retrieve the correct IDs, but when the BPA function attempts to process them by name, it generates an incorrect ID. What could be causing this issue?
![image](https://github.com/user-attachments/assets/fb74b12a-6aae-4808-b700-abecfd83732b)

![image](https://github.com/user-attachments/assets/0c68ff52-b9f4-48b7-95e7-586b0f208e61)

**Comment by m-kovalsky** [comment]
The next version of semantic link labs will have a function which explicitly does this - runs bpa for all semantic models in a workspace or set of workspaces and exports the results to a delta table. It makes sure to skip default semantic models as those are problematic. Look for 0.7.0 coming soon.

**Comment by pawarbi** [comment]
@matiirueda  Here is how I changed Romain's code for my purposes... labs will support this OOB so thats awesome, until then feel free to revise to your needs.


```
import pandas as pd
from datetime import datetime

WorkspaceSemanticModelName = "Sales"
df_datasets = fabric.list_datasets(WorkspaceSemanticModelName)
bpa_dfs = []
print(f"Workspace: {WorkspaceSemanticModelName} has {len(df_datasets)} datasets ")
for index, row in df_datasets.iterrows():
    mydataset = row['Dataset Name']
    if (mydataset != 'BPA_Lakehouse') and (mydataset != 'BPA Report'):
        try:
            df = labs.run_model_bpa(dataset=mydataset, workspace=WorkspaceSemanticModelName, return_dataframe=True)
            df['Workspace Name'] = WorkspaceSemanticModelName
            df['Dataset Name'] = mydataset
            df['Current Datetime'] = datetime.now()  # Add current datetime column
            bpa_dfs.append(df)
        except:
            continue

result = pd.concat(bpa_dfs)
# save it to a lakehouse
# spark.createDataFrame(result).write.format("delta").save("abfss path of the lakehouse")
```

**Comment by matiirueda** [comment]
> La próxima versión de los laboratorios de enlaces semánticos tendrá una función que explícitamente hace esto: ejecuta bpa para todos los modelos semánticos en un espacio de trabajo o conjunto de espacios de trabajo y exporta los resultados a una tabla delta. Se asegura de omitir los modelos semánticos predeterminados, ya que son problemáticos. Pronto llegará la versión 0.7.0.



Yes, exactly, I have been ignoring the default semantic models by name.

Do you have a date for the new release? Mainly to see if I should create a notebook per area or wait for this release.

**Comment by m-kovalsky** [comment]
Please try the run_model_bpa_bulk function available in [0.7.0](https://github.com/microsoft/semantic-link-labs/releases/tag/0.7.0).
